src/combinatorics/simple_graph/hasse.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import combinatorics.simple_graph.connectivity
+import data.fin.succ_pred
+import order.succ_pred.relation
+
+/-!
+# The Hasse diagram as a graph
```

Comment:
I think calling these Hasse diagrams is a stretch. They're certainly the underlying undirected graph of a Hasse diagram, but an important element of Hasse diagrams is that edges are directed so that you can (under suitable conditions) recover the order by taking the transitive closure. The `covby` relation itself could be considered to be a Hasse diagram.

My understanding of this module is that it's a systematic way to construct simple graphs from orders. Could we rename everything `hasse` to `from_covby`, or something like that?

Reply:
That's certainly less catchy. My idea was that `directed_graph.hasse` would be the directed version and `simple_graph.hasse` the undirected one.